### ISSUE TITLE
[UI Improvement] Make list of favorites accessible via search

### ISSUE THREAD

**Issue by scaidermern** [osmandapp/Osmand#6702]
It would be nice to get to the list of favorites in the search window. Searching for them is already possible, however scrolling through them still requires to navigate to a different menu item. I think it would be more intuitive to show them here, too.

Left: current UI. Right: suggested UI.

![photo_2019-03-18_09-50-36_both](https://user-images.githubusercontent.com/1040784/54518629-0d967700-4965-11e9-90da-07dad6e38fd3.jpg)

**Comment by scaidermern** [comment]
Alternatively if four tabs are too much then "Favorites" could become a new item in "Categories".

**Comment by sonora** [comment]
I am a little unclear what the _functional_  difference would be: Because both of today's "MyPlaces" items (favorites and tracks) are functionally included in the search anyway, any hits are sorted to the top of the hit list.

So are you voting to _duplicate_ the "Favorites" tab under "Search", or actually _move_ today's entire MyPlaces functionality under the roof of the Search screen?

**Comment by scaidermern** [comment]
Honestly I'm not so sure anymore about the use-case. Maybe this has something to do with searching for favorites being partly broken for a long time (#15291). Or with the fact that you have to search _by name_ instead of being able to just scroll through your favorites or tracks. If you created a list of places to visit during vacation you might not remember the name exactly, so being able to simply scroll through them would be nice. This is of course possible if you go to my places, but it isn't possible from the search window or when choosing a navigation target from the navigation menu.

**Comment by sonora** [comment]
Yes, I know there were a lot of bugs over time, quite a few resolved in the current v4.3.8.

Maybe we should close this issue, and re-evaluate, also light of the new "GPX tracks" UI which is in the making and has a cross impact... and then think about which use cases are left and still demand action?

**Comment by seniorm0ment** [comment]
Needing to scroll left and right for favorites is a big pain. The ability to view favorites in a vertical list, and sort it by name, search through just the favorites, and filter by last used, would be amazing.

**Comment by scaidermern** [comment]
> @scaidermern Regarding your original idea: Would it perhaps be a valid idea to entirely move the two tabs Favorites and Tracks from MyPlaces to the Search screen? In this case we would concentrate where people look for places **all on one screen**, which would visibly integrate both the favorites and tracks they have created themselves, as well as those searchable in offline maps, or online?

That's an interesting approach. But what about the other categories from My Places, mainly audio/video notes and OSM changes? Would those also become part of the search screen?

**Comment by sonora** [comment]
> Would those also become part of the search screen?

Yes  that was my thought.
- Put all this as tabs on one new "Places" screen.
- Have a Search govern all these on top, with its search results being displayed on each corresponding tab as the user selects/views one.

**Comment by scaidermern** [comment]
I guess this is a nice idea since it will make the UI more concise and easier to understand.

**Comment by vshcherb** [comment]
You still need to select tab to view results from the tab. One idea that could help to add to Categories: Favorites, Tracks, so there will be no need to type

**Comment by sonora** [comment]
>  One idea that could help to add to Categories: Favorites, Tracks, so there will be no need to type

I was really thinking about combining today's "My Places" and "Search" screens into one. As long as you do not type a search, it would fulfil the tab viewing / item management we today have on My Places. And it would also be your central entry point to search _anything_. (Already today we facilitate different searching on our Search screen and again for the MyPlaces tabs, so it seems rather disparsed...)

**Comment by vshcherb** [comment]
I don't see why do we combine screens Search / My Places. I think it's a bad UI practice cause you can't focus on most important action cause it could be different in different context. 
I see combined filter/search over my data (more functional might be) separate thing vs Search -> Favorites category

**Comment by sonora** [comment]
The idea comes from the observation that today we have de-central searches already on My Places > Favorites, on My Places Tracks, on Configure map > Tracks, and on the Search screen for its tabs History, Categories, Address. And potentially elsewhere. Often with a slightly different look and feel. So it could potentially be a beneficial and leaner concentration of all that functionality?
